descwl_coadd_task/_coadd_task.py

```diff
@@ -0,0 +1,509 @@
+        deferLoad=True,
+        multiple=True,
+    )
+    visitSummary = Input(
+        doc="Input visit-summary catalog with updated calibration objects.",
+        name="finalVisitSummary",
+        storageClass="ExposureCatalog",
+        dimensions=(
+            "instrument",
+            "visit",
+        ),
+        deferLoad=True,
+        multiple=True,
+    )
+    skyMap = Input(
+        doc="Input definition of geometry/bbox and projection/wcs for coadded "
+        "exposures",
+        name=BaseSkyMap.SKYMAP_DATASET_TYPE_NAME,
+        storageClass="SkyMap",
+        dimensions=("skymap",),
+    )
+    multiple_cell_coadd = Output(
+        doc="Output coadded exposure, produced by stacking input warps",
+        name="shearCoadd",
+        storageClass="MultipleCellCoadd",
+        dimensions=("tract", "patch", "skymap", "band"),
+    )
+
+    def __init__(self, *, config=None):
+        super().__init__(config=config)  # Harmless and a no-op for now.
+        if not config:
+            return
+
+        # Dynamically set input connections for noise images depending on the
+        # number of noise realizations specified in the config.
+        for n in range(config.num_noise_realizations):
+            noise_warps = Input(
+                doc="Input noise warp to be coadded.",
+                name=f"{config.connections.coaddType}calexp_noise{n}_descwarp",
+                storageClass="MaskedImageF",
+                dimensions=("tract", "patch", "skymap", "visit", "instrument"),
+                deferLoad=True,
+                multiple=True,
+            )
+            setattr(self, f"noise{n}_warps", noise_warps)
+
+
+class AssembleShearCoaddConfig(
+    PipelineTaskConfig, pipelineConnections=AssembleShearCoaddConnections
+):
+    """Configuration for AssembleShearCoaddTask."""
+
+    bad_mask_planes = ListField[str](
+        doc="Mask planes that, if set, the corresponding pixel would not be "
+        "included in the coadd.",
+        default=FLAGS2INTERP,
+    )
+    psf_dimensions = Field[int](
+        default=21,
+        doc="Dimensions of the PSF image stamp size to be assigned to cells.",
+        # Check that psf_dimensions is positive and odd.
+        check=lambda x: (x > 0) and (x % 2 == 1),
+    )
+    psf_warper = ConfigField(
+        dtype=afwMath.Warper.ConfigClass,
+        doc="Configuration for the warper that warps the PSF."
+        "This must match the configuration used to warp the images.",
+    )
+    max_maskfrac = RangeField[float](
+        default=0.9,
+        min=0.0,
+        inclusiveMin=True,
+        max=1.0,
+        inclusiveMax=True,
+        doc="Maximum fraction of masked pixels to allow. Warps with maskfrac "
+        "values greater than or equal to this value within the cell's "
+        "bounding box (outer) are excluded from the coadd. "
+        "Values must be in the range [0, 1].",
+    )
+    num_noise_realizations = Field[int](
+        default=1,
```

Comment:
```suggestion
        default=3,
```

We want at least 3.

Reply:
True, but let's come back to this when we actually need it with metadetect. For running coadds and doing initial round of validation, this should be sufficient.